vizro-core/tests/unit/vizro/models/_navigation/test_navigation.py

```diff
@@ -11,51 +12,59 @@
-        with pytest.warns(UserWarning):
-            vm.Navigation(pages=["Page 1"])
+    def test_navigation_valid_pages_not_all_included(self):
+        navigation = vm.Navigation(pages=["Page 1"], id="navigation")
+        assert navigation.id == "navigation"
+        assert navigation.pages == ["Page 1"]
+
+    def test_navigation_invalid_pages_empty_list(self):
+        with pytest.raises(ValidationError, match="Ensure this value has at least 1 item."):
+            vm.Navigation(pages=[], id="navigation")
+
+    def test_navigation_invalid_pages_unknown_page(self):
+        with pytest.raises(ValidationError, match=re.escape("Unknown page ID ['Test'] provided to argument 'pages'.")):
+            vm.Navigation(pages=["Test"], id="navigation")
 
-    @pytest.mark.parametrize(
-        "pages, expected_error",
-        [
-            ([], "Ensure this value has at least 1 item."),
-            ([Accordion()], "2 validation errors for Navigation"),
-            (["Page_1", "Page 2"], "1 validation error for Navigation"),
-        ],
-    )
-    def test_field_invalid_pages_input(self, pages, expected_error):
-        with pytest.raises(ValidationError, match=expected_error):
-            vm.Navigation(pages=pages)
+
+@pytest.mark.usefixtures("dashboard_prebuild")
+@pytest.mark.parametrize("pages", [["Page 1", "Page 2"], {"SELECT PAGE": ["Page 1", "Page 2"]}])
+def test_navigation_pre_build(pages):
+    navigation = vm.Navigation(pages=pages, id="navigation")
+    navigation.pre_build()
+
+    assert navigation.id == "navigation"
+    assert navigation.pages == pages
+    assert isinstance(navigation._selector, Accordion)
+    assert navigation._selector.pages == {"SELECT PAGE": ["Page 1", "Page 2"]}
 
 
 @pytest.mark.usefixtures("dashboard_prebuild")
-class TestNavigationBuild:
-    """Tests navigation build method."""
-
-    @pytest.mark.parametrize("pages", [["Page 1", "Page 2"], {"Page 1": ["Page 1"], "Page 2": ["Page 2"]}, None])
-    def test_navigation_build(self, pages):
-        navigation = vm.Navigation(pages=pages)
-        navigation.pre_build()
-        accordion = Accordion(pages=pages)
-        navigation._selector.id = accordion.id
-
-        result = json.loads(json.dumps(navigation.build(), cls=plotly.utils.PlotlyJSONEncoder))
-        expected = json.loads(json.dumps(accordion.build(), cls=plotly.utils.PlotlyJSONEncoder))
-        assert result == expected
+@pytest.mark.parametrize("pages", [["Page 1", "Page 2"], {"Page 1": ["Page 1"], "Page 2": ["Page 2"]}])
+def test_navigation_build(pages):
+    navigation = vm.Navigation(pages=pages)
+    navigation.pre_build()  # Required such that an Accordion is assigned as selector
+    accordion = Accordion(pages=pages)
+    navigation._selector.id = accordion.id
+
+    result = json.loads(json.dumps(navigation.build(), cls=plotly.utils.PlotlyJSONEncoder))
+    expected = json.loads(json.dumps(accordion.build(), cls=plotly.utils.PlotlyJSONEncoder))
```

Comment:
I know this was already previously there, but are finding it ok to skip the direct object comparison, because the build of the `Accordion` has already been explicityl tested before?

Reply:
Yes, we shouldn't have to test it again - what is important here is that the navigation.build() returns the build of it's selector. It's very similar to our tests for Filters, where in the build we also only that it is the same as the selector.build